nrf_wifi/fw_if/umac_if/src/fmac_api_common.c

```diff
@@ -849,6 +855,105 @@ enum nrf_wifi_status nrf_wifi_fmac_get_reg(struct nrf_wifi_fmac_dev_ctx *fmac_de
 	return NRF_WIFI_STATUS_FAIL;
 }
 
+int nrf_wifi_phy_rf_params_init(struct nrf_wifi_osal_priv *opriv,
+				struct nrf_wifi_phy_rf_params *prf,
+				unsigned char package_info,
+				unsigned char *hex_arr,
+				unsigned char *str)
+{
+	int ret = -1;
+	/* Initilaize reserved bytes */
+	nrf_wifi_osal_mem_set(opriv,
+			      &prf->reserved,
+			      0x0,
+			      sizeof(prf->reserved));	
+	/* Initialize PD adjust values for MCS7. Currently these 4 bytes are not being used */
+	prf->pd_adjust_val.pd_adjt_lb_chan = PD_ADJUST_VAL;
+	prf->pd_adjust_val.pd_adjt_hb_low_chan = PD_ADJUST_VAL;
+	prf->pd_adjust_val.pd_adjt_hb_mid_chan = PD_ADJUST_VAL;
+	prf->pd_adjust_val.pd_adjt_hb_high_chan = PD_ADJUST_VAL;
+
+	/* Configure systematic offset value */
+	prf->syst_tx_pwr_offset.syst_off_lb_chan = SYSTEM_OFFSET_LB;
+	prf->syst_tx_pwr_offset.syst_off_hb_low_chan = SYSTEM_OFFSET_HB_CHAN_LOW;
+	prf->syst_tx_pwr_offset.syst_off_hb_mid_chan = SYSTEM_OFFSET_HB_CHAN_MID;
+	prf->syst_tx_pwr_offset.syst_off_hb_high_chan = SYSTEM_OFFSET_HB_CHAN_HIGH;
+
+	/* RX Gain offsets */
+	prf->rx_gain_offset.rx_gain_lb_chan = RX_GAIN_OFFSET_LB_CHAN;
+	prf->rx_gain_offset.rx_gain_hb_low_chan = RX_GAIN_OFFSET_HB_LOW_CHAN;
+	prf->rx_gain_offset.rx_gain_hb_mid_chan = RX_GAIN_OFFSET_HB_MID_CHAN;
+	prf->rx_gain_offset.rx_gain_hb_high_chan = RX_GAIN_OFFSET_HB_HIGH_CHAN;
+
+	nrf_wifi_osal_mem_set(opriv,
```

Comment:
Is it better to initialize the entire structure to `0xFF` in case the caller of this function passes a truncated `str` ?